lectures/12-plonk.tex

```diff
@@ -0,0 +1,234 @@
+
+\[S(y) \cdot [T(y) + T(\omega y)] + (1 - S(y)) \cdot T(y) \cdot T(\omega y) = T(\omega^2 y)\]
+where \(T(y)\) and \(T(\omega y)\) are left and right inputs correspondingly.
+
+This means, that once again we can narrow our check down to \textit{ZeroTest} for \(\forall \, y \in \Omega_{\text{gates}}\):
+
+\[S(y) \cdot [T(y) + T(\omega y)] + (1 - S(y)) \cdot T(y) \cdot T(\omega y) - T(\omega^2 y) = 0\]
+
+\begin{example}
+
+That is, in our \(C\), since gates 0 and 1 are addition and 2 gate is multiplication, we have the following encoding for selector polynomial:
+
+\begin{tabular}{l ccc | c}
+  inputs: & 5, & 6, & 1 & $S(X)$ \\ \hline
+  Gate 0 ($\omega^0$): & 5, & 6, & 11 & 1 \\
+  Gate 1 ($\omega^3$): & 6, & 1, & 7 & 1 \\
+  Gate 2 ($\omega^6$): & 11, & 7, & 77 & 0 \\ 
+\end{tabular}
+
+You can see by substituting \(S(y)\) with actual values from the table, that if a selector polynomial evaluates as \(1\), then multiplication part of the proved constraint is leveling out - vice versa for \(0\).
+
+\end{example}
+
+\subsubsection{The wiring is implemented correctly}
+In this part, we need to encode the wires of \(C\) to prove connection of inputs and outputs in gates. 
+
+\begin{example}
+For our table:
+\begin{tabular}{l ccc}
+  $\omega^{-1}$, $\omega^{-2}$, $\omega^{-3}$: & 5, & \textcolor{violet}{6}, & \textcolor{red}{1} \\ \hline
+  0: $\omega^{0}$, $\omega^{1}$, $\omega^{2}$: & 5, & \textcolor{violet}{6}, & \textcolor{green}{11} \\
+  1: $\omega^{3}$, $\omega^{4}$, $\omega^{5}$: & \textcolor{violet}{6}, & \textcolor{red}{1}, & \textcolor{gray}{7} \\
+  2: $\omega^{6}$, $\omega^{7}$, $\omega^{8}$: & \textcolor{green}{11}, & \textcolor{gray}{7}, & 77 \\ 
+\end{tabular}
+\\
+We need following constraints:
+\begin{align}
+    T(\omega^{-2}) &= T(\omega^{1}) = T(\omega^{3}) \\
+    T(\omega^{-1}) &= T(\omega^{0}) \\
+    T(\omega^{2})  &= T(\omega^{6}) \\
+    T(\omega^{-3}) &= T(\omega^{4})
+\end{align}
+
+For that matter, define a polynomial \(W = \Omega \to \Omega\) that implements a rotation:
+
+\[W(\omega^{-2}, \omega^{1}, \omega^{3}) = (\omega^{1}, \omega^{3}, \omega^{-2}), \quad W(\omega^{-1}, \omega^{0}) = (\omega^{0}, \omega^{-1}), \quad \dots\]
+
+\end{example}
+
+\begin{lemma}
+Lemma: \(\forall \, y \in \Omega : T(y) = T(W(y)) =>\) wire constraints are satisfied. This may be proven using \textit{prescribed permutation check}.
+\end{lemma}
+
+\subsubsection{Output of the last gate conforms with expected}
+
+Let \(\alpha\) be the expected output of \(C\). Then, apply \textit{ZeroTest} on:
+\[T(\omega^{3 |C| - 1}) - \alpha = 0\]
+
+\subsection{Setup procedure}
+
+Not accounting for the details of selected poly-commit scheme (i.e KZG, FRI, etc.) the setup procedure preprocesses the circuit \(C\) and outputs for the prover selector and wiring polynomials \(S\) and \(W\), and for the verifier respective commitments \([S]\) and \([W]\).
+
+\begin{remark}
+This setup procedure is untrusted.
+\end{remark}
+
```

Comment:
After that, I would like to see the whole protocol written in a single "Protocol box", similarly to the previous section